src/Package/Test/ProjectSystem/PropertyPages/ProjectSettingsControlTest.cs

```diff
@@ -0,0 +1,255 @@
+
+            var folder = Path.GetTempPath();
+            var fileName1 = "0123.Settings.R";
+            var file1 = Path.Combine(folder, fileName1);
+            var fileName2 = "0456.Settings.R";
+            var file2 = Path.Combine(folder, fileName2);
+            try {
+                using (var sw = new StreamWriter(file1)) {
+                    sw.WriteLine("x <- 1");
+                }
+                using (var sw = new StreamWriter(file2)) {
+                    sw.WriteLine("y <- 2");
+                }
+                fs.GetFileSystemEntries(Arg.Any<string>(), Arg.Any<string>(), SearchOption.AllDirectories).Returns(new string[] { file1, file2 });
+
+                control.SetProjectAsync(folder, null);
+
+                control.CreateControl();
+                control.FileListCombo.Items.Count.Should().BeGreaterThan(0);
+                control.FileListCombo.Items.Should().Contain("~/" + fileName1);
+                control.FileListCombo.Items.Should().Contain("~/" + fileName2);
+
+                control.PropertyGrid.SelectedObject.Should().NotBeNull();
+                var desc = control.PropertyGrid.SelectedObject as SettingsTypeDescriptor;
+                desc.Should().NotBeNull();
+                desc.GetProperties().Should().HaveCount(1);
+
+                var prop = desc.GetProperties()[0] as SettingPropertyDescriptor;
+                prop.Should().NotBeNull();
+                prop.SetValue(null, "42");
+                prop.Setting.Value.Should().Be("42");
+
+                var pg = control.PropertyGrid;
+                var mi = pg.GetType().GetMethod("OnPropertyValueChanged", BindingFlags.Instance | BindingFlags.NonPublic);
+                mi.Invoke(pg, new object[] { new PropertyValueChangedEventArgs(pg.SelectedGridItem, "1") });
+
+                control.IsDirty.Should().BeTrue();
+                shell.ShowMessage(Microsoft.VisualStudio.R.Package.Resources.SettingsPage_SavePrompt, MessageButtons.YesNoCancel).Returns(MessageButtons.Cancel);
+
+                control.FileListCombo.SelectedIndex = 1;
+                shell.Received().ShowMessage(Microsoft.VisualStudio.R.Package.Resources.SettingsPage_SavePrompt, MessageButtons.YesNoCancel);
+
+                control.IsDirty.Should().BeTrue();
+                control.FileListCombo.SelectedIndex.Should().Be(0);
+            } finally {
+                if (File.Exists(file1)) {
+                    File.Delete(file1);
+                }
+                if (File.Exists(file2)) {
+                    File.Delete(file2);
+                }
+            }
+        }
+
+        [Test]
+        public async Task PropertyGridMultiple02() {
+            var shell = Substitute.For<ICoreShell>();
+            var fs = Substitute.For<IFileSystem>();
+            var control = new SettingsPageControl(shell, fs);
+
+            var folder = Path.GetTempPath();
+            var fileName1 = "1123.Settings.R";
```

Comment:
There is no general `TestFilesFixture` in package tests, but you can create your own deriving it from `DeployFilesFixture`.